Fixed ACS HudMessages not working properly online, and unified all PrintHUDMessage server commands

AKMDM opened the pull request:
This pull request fixes the following issues on the Zandronum tracker: https://zandronum.com/tracker/view.php?id=3200 and https://zandronum.com/tracker/view.php?id=3623 so everything with ACS HudMessages should now work properly in online games. This unifies all the `PrintHUDMessage` server commands into one single command, which removes a lot of duplicate code inside `cl_main.cpp`, `sv_main.cpp`, and `sv_commands.cpp`. By combining these commands into one, it reduces the total number of `SVC` commands we have defined, which is significant because in https://github.com/TorrSamaho/zandronum/pull/6, four new server commands that fix line texture offsets and scale need to be added too, so the difference should almost cancel out.

I also tried optimizing HudMessages wherever I thought was necessary. For example, if the width and the height of the HUD are both passed as zero, then they aren't sent to the clients. Likewise, the name of the font won't be sent to the clients if it's `SmallFont`, which is also the default font of any ACS script. Clipping rectangles and alpha for ACS HUD messages won't be sent either, unless needed. The HUD message types (`NORMAL`, `FADEOUT`, `TYPEON`, and `FADEINOUT`) are passed along with some flags (e.g. `ALPHA`, `ADDBLEND`, `LOG`, `NOWRAP`).

TorrSamaho commented on src/duel.cpp:
```diff
@@ -411,7 +411,7 @@ void DUEL_TimeExpired( void )
 			else
 			{
-				SERVERCOMMANDS_PrintHUDMessageFadeOut( szString, 160.4f, 75.0f, 320, 200, CR_RED, 3.0f, 2.0f, "BigFont", false, MAKE_ID('C','N','T','R') );
+				SERVERCOMMANDS_PrintHUDMessage( szString, 160.4f, 75.0f, 320, 200, HUDMESSAGETYPE_FADEOUT, CR_RED, 3.0f, NULL, 2.0f, "BigFont", MAKE_ID( 'C', 'N', 'T', 'R' ) );
```
The `NULL` is the `float fInTime` argument, right? Then, I'd put `0.0f` instead. Same applies to all other occasions. `NULL` seems to imply that you want to pass a NULL pointer.

AKMDM replied:
I will change that (I thought by using `NULL` it would better convey that this argument wasn't being used, sorry for the confusion), thank you for correcting me on this.

TorrSamaho replied:
No problem and thanks for changing!

TorrSamaho commented on src/sv_commands.cpp:
```diff
@@ -2070,78 +2070,92 @@ void SERVERCOMMANDS_PrintMOTD( const char *pszString, ULONG ulPlayerExtra, Serve
-	command.SetId( lID );
+
+	// [AK] Initialize all unused members.
+	command.SetClipRectLeft( NULL );
```
No of these `NULL`s are pointer argument, right? Then I'd put the corresponding zero numbers here to point out what the intended argument type is.

TorrSamaho commented on protocolspec/spec.print.txt:
```diff
@@ -16,55 +16,49 @@ Command PrintHUDMessage
 	Float y
-	Short hudWidth
-	Short hudHeight
+	Long type
```
Do we really need a `Long` here? Shouldn't a `Short` be sufficient if you move `HUDMESSAGE_SEND_ALPHA` and `HUDMESSAGE_DONTSEND_WRAP` to lower bytes?

AKMDM replied:
I can try doing this. I might need to create new definitions for some of the flags, particularly `HUDMSG_ADDBLEND` and `HUDMSG_LOG` so they all don't occupy more than 16 bits. The four different HudMessage types will only ever occupy the first 2 bits, so this could work.

AKMDM commented:
I added changes to this pull request, based on the comments from last week. All those `NULL` keywords have now been replaced with zeros, and I was able to fit all the needed HUDMessage types and flags into a single 16-bit integer, and changed `type` from a `Long` into a `Short`, thereby saving two bytes of bandwidth for each call.

I did have to redefine a lot of these flags so they could fit on smaller bits, but I added some comments to help explain what's happening.

TorrSamaho commented:
Thanks for the updates! Looks good to me now. I compressed, rebased and added this to the head.
